pkg/dynamicinstrumentation/codegen/c/maps.h

```diff
@@ -0,0 +1,41 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+#ifndef DI_MAPS_H
+#define DI_MAPS_H
+
+struct {
```

Comment:
add a comment to explain the purpose of each map

Reply:
also why not use [map-defs.h](https://github.com/DataDog/datadog-agent/blob/ed7ba4e5d3298b026fc96173c3f8d617d3995946/pkg/ebpf/c/map-defs.h) macros?

Reply:
I've updated to use the macros in map-defs.h, and had to adapt the macros a bit to support stacks which would fail on creation if `key` is specified. Please take a look at that commit :+1: